src/tape_drivers/netbsd/scsipi-ibmtape/scsipi_scsi_tape.c

```diff
@@ -0,0 +1,326 @@
+		rc = _sense2errorcode(sense_value, vendor_table, msg, MASK_WITH_SENSE_KEY);
+
+	return rc;
+}
+
+static bool is_expected_error(struct scsipi_tape *device, uint8_t *cdb, int32_t rc )
+{
+	int cmd = (cdb[0]&0xFF);
+	uint64_t destination;
+	uint64_t cdb_dest[8];
+	int i;
+
+	switch (cmd) {
+		case TEST_UNIT_READY:
+			if (rc == -EDEV_NEED_INITIALIZE || rc == -EDEV_CONFIGURE_CHANGED)
+				return true;
+			break;
+		case READ:
+			if (rc == -EDEV_FILEMARK_DETECTED || rc == -EDEV_NO_SENSE || rc == -EDEV_CLEANING_REQUIRED)
+				return true;
+			if ((rc == -EDEV_CRYPTO_ERROR || rc == -EDEV_KEY_REQUIRED) && !device->is_data_key_set)
+				return true;
+			break;
+		case WRITE:
+			if (rc == -EDEV_EARLY_WARNING || rc == -EDEV_PROG_EARLY_WARNING || rc == -EDEV_CLEANING_REQUIRED)
+				return true;
+			break;
+		case WRITE_FILEMARKS6:
+			if (rc == -EDEV_EARLY_WARNING || rc == -EDEV_PROG_EARLY_WARNING || rc == -EDEV_CLEANING_REQUIRED)
+				return true;
+			break;
+		case LOAD_UNLOAD:
+			if ((cdb[4] & 0x01) == 0) // Unload
+				if (rc == -EDEV_CLEANING_REQUIRED)
+					return true;
+			break;
+		case MODE_SELECT10:
+			if (rc == -EDEV_MODE_PARAMETER_ROUNDED)
+				return true;
+			break;
+		case LOCATE16:
+			for (i=0; i<8; i++)
+				cdb_dest[i] = (uint64_t)cdb[i+4] & 0xff;
+
+			destination = (cdb_dest[0] << 56) + (cdb_dest[1] << 48)
+			+ (cdb_dest[2] << 40) + (cdb_dest[3] << 32)
+			+ (cdb_dest[4] << 24) + (cdb_dest[5] << 16)
+			+ (cdb_dest[6] << 8) + cdb_dest[7];
+
+			if (destination == TAPE_BLOCK_MAX && rc == -EDEV_EOD_DETECTED)
+				return true;
+			break;
+	}
+
+	return false;
+}
+
+/* Global functions */
+#define HOST_OK          (0x00)
+#define HOST_NO_CONNECT  (0x01)
+#define HOST_BUS_BUSY    (0x02)
+#define HOST_TIME_OUT    (0x03)
+#define HOST_BAD_TARGET  (0x04)
+#define HOST_ABORT       (0x05)
+#define HOST_PARITY      (0x06)
+#define HOST_ERROR       (0x07)
+#define HOST_RESET       (0x08)
+#define HOST_BAD_INTR    (0x09)
+#define HOST_PASSTHROUGH (0x0a)
+#define HOST_SOFT_ERROR  (0x0b)
+#define HOST_IMM_RETRY   (0x0c)
+#define HOST_REQUEUE     (0x0d)
+#define HOST_TRANS_DISR  (0x0e)
+#define HOST_TRANS_FAIL  (0x0f)
+#define HOST_TARGET_FAIL (0x10)
+#define HOST_NEXUS_FAIL  (0x11)
+
+#define DRIVER_OK        (0x00)
+#define DRIVER_BUSY      (0x01)
+#define DRIVER_SOFT      (0x02)
+#define DRIVER_MEDIA     (0x03)
+#define DRIVER_ERROR     (0x04)
+#define DRIVER_INVALID   (0x05)
+#define DRIVER_TIMEOUT   (0x06)
+#define DRIVER_HARD      (0x07)
+#define DRIVER_SENSE     (0x08)
+
+#define NO_SUGGESTION    (0x00)
+#define SUGGEST_RETRY    (0x10)
+#define SUGGEST_ABORT    (0x20)
+#define SUGGEST_REMAP    (0x30)
+#define SUGGEST_DIE      (0x40)
+#define SUGGEST_SENSE    (0x80)
+
+int scsipi_issue_cdb_command(struct scsipi_tape *device, scsireq_t *req, 
+			 char *desc, char **msg)
+{
+	int status = 0;
+	int ret = -1;
+	
+	CHECK_ARG_NULL(req, -LTFS_NULL_ARG);
+	CHECK_ARG_NULL(msg, -LTFS_NULL_ARG);
+
+fprintf(stderr, "CDB [ ");
```

Comment:
Normally `ltfs` process runs on background, fuse_main() daemonize `ltfs` automatically except with `-f` option. So print out to `stderr` is not a good idea.

Could you remove this debug purpose code?  I believe this is debug purpose, otherwise could you tell me why this output is required?